### PR TITLE
mariadb: remove withoutClient

### LINKED ISSUE #82428: nixos/mysql: Error first init mysql databases
**Describe the bug**
Erorr  first init mysql databases if use package mariadb.server builded `withoutClient = true`

**To Reproduce**
1. Apply this patch
```
diff --git a/nixos/tests/mysql.nix b/nixos/tests/mysql.nix
index 924bac84e26..a4a245f4741 100644
--- a/nixos/tests/mysql.nix
+++ b/nixos/tests/mysql.nix
@@ -26,6 +26,13 @@ import ./make-test-python.nix ({ pkgs, ...} : {
       { pkgs, ... }:

       {
+        nixpkgs.overlays = [
+          (self: super: {
+            mariadb = super.mariadb.override {
+              withoutClient = true;
+            };
+          })
+        ];
         users.users.testuser = { };
         users.users.testuser2 = { };
         services.mysql.enable = true;

```
2. run command `nix-build nixos/tests/mysql.nix`

**Expected behavior**
Error running this command
```
cat /nix/store/36wz42cljwgx94kba5pgfavrkjw57g61-mysql_init.sql | /nix/store/9zjrpl0blyi7rzlzsd2nx4pjjacgpybx-mariadb-server-10.3.22/bin/mysql -u root -N
    rm /tmp/mysql_init
fi
```
```
l /nix/store/9zjrpl0blyi7rzlzsd2nx4pjjacgpybx-mariadb-server-10.3.22/bin/mysql
ls: невозможно получить доступ к '/nix/store/9zjrpl0blyi7rzlzsd2nx4pjjacgpybx-mariadb-server-10.3.22/bin/mysql': No such file or directory

```

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.5.9-hardened, NixOS, 20.09.git.ee6a3c3bd0a (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(root): `"nixos-20.09pre216451.ce961fad160"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

cc @aanderse

### PR DESCRIPTION
**Pull request by flokli**
When used as a global override, it breaks most of the options in the
mysql module, such as ensureDatabases, ensureUsers, initialDatabases,
initialScript.

We could use `.client` there, but if the reasoning behind this was
closure size reduction, we now end up with the same (or a bigger)
runtime closure and more complexity.

Apart from the options exposed by the mysql module, the client is also
likely to be required for local backups or DBA tasks anyways.

Instead of dealing with all the increased complexity of this for no
arguable benefit, let's just remove the `withoutClient` argument.
Storage space on mysql servers shouldn't be that much of an issue.

Closes #82428.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
@flokli need remove `cmake-without-client.patch`

File: pkgs/servers/sql/mariadb/default.nix [hunk 1]
```diff
@@ -185,14 +180,6 @@ server = stdenv.mkDerivation (common // {
     chmod +x "$out"/bin/wsrep_sst_common
     rm "$out"/bin/{mysql_client_test,mysqltest}
     rm -r "$out"/data # Don't need testing data
-  '' + optionalString withoutClient ''
-    ${ # We don't build with GSSAPI on Darwin
-      optionalString (!stdenv.hostPlatform.isDarwin) ''
-        rm "$out"/lib/mysql/plugin/auth_gssapi_client.so
-      ''
-    }
```
Comment (Izorkin) [inline pkgs/servers/sql/mariadb/default.nix]: This code dont need remove.
https://github.com/NixOS/nixpkgs/pull/63355/commits/c8488b913a285b72bb9dc090f2903762b11ac9ce

Comment (flokli) [reply pkgs/servers/sql/mariadb/default.nix]: Yes it does - the whole thing is wrapped inside a `optionalString withoutClient` - so everything inside needs to go.

Comment (Izorkin) [reply pkgs/servers/sql/mariadb/default.nix]: Checked with this patch
```
diff --git a/pkgs/servers/sql/mariadb/default.nix b/pkgs/servers/sql/mariadb/default.nix
index 51181c4d2ae..da173b8b678 100644
--- a/pkgs/servers/sql/mariadb/default.nix
+++ b/pkgs/servers/sql/mariadb/default.nix
@@ -4,7 +4,6 @@
 , fixDarwinDylibNames, cctools, CoreServices, less
 , numactl # NUMA Support
 , withStorageMroonga ? true, kytea, msgpack, zeromq
-, withoutClient ? false
 }:

 with stdenv.lib;
@@ -149,9 +148,7 @@ server = stdenv.mkDerivation (common // {
     ++ optional stdenv.hostPlatform.isLinux linux-pam
     ++ optional (!stdenv.hostPlatform.isDarwin) mytopEnv;

-  patches = common.patches ++ [
-    ./cmake-without-client.patch
-  ] ++ optionals stdenv.hostPlatform.isDarwin [
+  patches = common.patches ++ optionals stdenv.hostPlatform.isDarwin [
     ./cmake-without-plugin-auth-pam.patch
   ];

@@ -170,8 +167,6 @@ server = stdenv.mkDerivation (common // {
     "-DWITH_NUMA=ON"
   ] ++ optional (!withStorageMroonga) [
     "-DWITHOUT_MROONGA=ON"
-  ] ++ optionals withoutClient [
-    "-DWITHOUT_CLIENT=ON"
   ] ++ optionals stdenv.hostPlatform.isDarwin [
     "-DWITHOUT_OQGRAPH=1"
     "-DWITHOUT_TOKUDB=1"
@@ -185,14 +180,11 @@ server = stdenv.mkDerivation (common // {
     chmod +x "$out"/bin/wsrep_sst_common
     rm "$out"/bin/{mysql_client_test,mysqltest}
     rm -r "$out"/data # Don't need testing data
-  '' + optionalString withoutClient ''
     ${ # We don't build with GSSAPI on Darwin
       optionalString (!stdenv.hostPlatform.isDarwin) ''
         rm "$out"/lib/mysql/plugin/auth_gssapi_client.so
       ''
     }
-    rm "$out"/lib/mysql/plugin/client_ed25519.so
-    rm "$out"/lib/{libmysqlclient${libExt},libmysqlclient_r${libExt}}
   '' + optionalString withStorageMroonga ''
     mv "$out"/share/{groonga,groonga-normalizer-mysql} "$out"/share/doc/mysql
   '' + optionalString (!stdenv.hostPlatform.isDarwin) ''

```

Comment (flokli) [reply pkgs/servers/sql/mariadb/default.nix]: `withoutClient` defaults to false, meaning, everything from https://github.com/NixOS/nixpkgs/pull/82649/files#diff-6967ff0d62d213fc6cd7c7d8207e3f11L188 till https://github.com/NixOS/nixpkgs/pull/82649/files#diff-6967ff0d62d213fc6cd7c7d8207e3f11L195 is always false and can be removed.

Your proposed patch would now still remove `$out"/lib/mysql/plugin/auth_gssapi_client.so` on non-darwin, even though we didn't do that before (as it was guarded inside  the `optionalString withoutClient`)

Comment (Izorkin) [reply pkgs/servers/sql/mariadb/default.nix]: > Your proposed patch would now still remove `$out"/lib/mysql/plugin/auth_gssapi_client.so` on non-darwin, even though we didn't do that before (as it was guarded inside the `optionalString withoutClient`)

https://github.com/NixOS/nixpkgs/commit/b8a6649797a73f75586c589a1b4c8197d9920c43#diff-6967ff0d62d213fc6cd7c7d8207e3f11R171-R177

Comment (Izorkin) [reply pkgs/servers/sql/mariadb/default.nix]: Sorry, dont need remove file `auth_gssapi_client.so`

**Review by Izorkin — APPROVED** [review]

**Pull request merged**